src/lib/fsm/index.ts

```diff
@@ -0,0 +1,54 @@
+export default class fsm {
+  private currentState: string;
+  private states: { [id: string]: { [id: string]: string } } = {};
```

Comment:
specify types, rather than: `{ [id: string]: { [id: string]: string } }` use a type.
Use a Map https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Map

conventions: (Unspecified) I would always initialize on ctor, rather than default initialization on field definition.